### PR TITLE
Refactor the registries to contain data relevant to the target.

### PR DESCRIPTION
**Pull request by beeperdeeper089**
Resolves: #
Authors:
@beeperdeeper089 

## Summary of changes
- Replace Common/BlockRegistry with BlockReferrer.
- Update all relevant classes with the correct things
- Make a BlockRegistry on Client and Server which contains the data only they need.
  
## Caveats
Does this introduce any new bugs?
- Currently nothing works cos of a bug on develop.

## On approval
merge

### CODE REVIEW

File: Phoenix/Client/Include/Client/Voxels/BlockRegistry.hpp [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+#include <Common/CMS/ModManager.hpp>
+#include <Common/Registry.hpp>
+
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+namespace phx::client
+{
+	struct BlockRegistry
```
Comment (SuperFola) [inline Phoenix/Client/Include/Client/Voxels/BlockRegistry.hpp:41]: Doc?

File: Phoenix/Client/Include/Client/Voxels/BlockRegistry.hpp [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+	{
+		voxels::BlockReferrer referrer;
+
+		Registry<std::size_t, std::vector<std::string>> textures;
+
+		void registerAPI(cms::ModManager* manager)
```
Comment (SuperFola) [inline Phoenix/Client/Include/Client/Voxels/BlockRegistry.hpp:47]: Doc?

File: Phoenix/Common/Include/Common/Voxels/BlockReferrer.hpp [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+#pragma once
+
+#include <Common/Voxels/Block.hpp>
+#include <Common/Registry.hpp>
+
+#include <string>
+
+namespace phx::voxels
+{
+	struct BlockReferrer
```
Comment (SuperFola) [inline Phoenix/Common/Include/Common/Voxels/BlockReferrer.hpp:38]: Doc?

**Comment by SuperFola** [comment]
Will merge once checks are done running

**Comment by tobyplowy** [comment]
Merging as checks have passed.

**Pull request merged**